src/browser-lib/cookie-consent/CookieConsentProvider.tsx

```diff
@@ -24,55 +20,17 @@ import {
 
 import { consentClient } from "./consentClient";
 
-export type CookieConsentContext = {
-  /**
-   * Show the cookie consent modal
-   */
-  showConsentManager: () => void;
-  /**
-   * Get the current consent state for a given policy
-   */
-  getConsentState: (
-    policyName: "strictly-necessary" | "embedded-content" | "statistics",
-  ) => "granted" | "denied" | "pending";
-};
-
-export const cookieConsentContext = createContext<CookieConsentContext | null>(
-  null,
-);
-
-/**
- * Provides methods to open the consent settings and get the current consent state for a policy
- */
-export const useCookieConsent = () => {
-  const context = useContext(cookieConsentContext);
-
-  if (!context) {
-    throw new Error(
-      "useCookieConsent() called outside of cookieConsentContext provider",
-    );
-  }
-
-  return context;
-};
-
-const CookieConsentContextProvider = (props: PropsWithChildren) => {
-  const { state, getConsent: getConsentState } = useOakConsent();
-  const { showBanner, openSettings: showConsentManager } = useCookieConsentUI();
+const RequiresInteraction = () => {
+  const { state } = useOakConsent();
+  const { showBanner } = useCookieConsentUI();
 
   useEffect(() => {
     if (state.requiresInteraction) {
       showBanner();
     }
   }, [state.requiresInteraction, showBanner]);
 
-  return (
-    <cookieConsentContext.Provider
-      {...props}
-      // This value should not be memoised as it should always trigger a render whenever `state` changes
-      value={{ getConsentState, showConsentManager }} //NOSONAR
-    />
-  );
```

Comment:
Now that its easy to mock the consent client we can use the hooks from oak-consent-client and oak-components directly making this provider redundant